chrome/content/zotero/xpcom/data/items.js

```diff
@@ -625,6 +625,47 @@ Zotero.Items = function() {
 	});
 	
 	
+	/**
+	 * Returns an array of items with groups of child items replaced with parents
+	 * where possible.
+	 * @return {Array<Zotero.Item>}
+	 */
+	this.replaceChildren = function(items) {
+		var parents = {}, itemMap = {};
+		for (let item of items) {
+			itemMap[item.id] = item;
+			if (item.parentID) {
+				// Found a new parent
+				if (! parents[item.parentID]) {
+					parents[item.parentID] = [this.get(item.parentID), []]
+				}
+				let parent = parents[item.parentID];
+				// Track all child items of a parent
+				parent[1].push(item);
+			}
+		}
+		for (let id in parents) {
+			// If all children of a parent already in items OR
+			// parent itself is already in the array
+			// replace with just the parent and remove the children.
+			let parent = parents[id];
+			if(parent[0].numChildren() == parent[1].length || itemMap[parent[0].id]) {
+				for (let i of parent[1]) {
+					delete itemMap[i.id];
+				}
+				itemMap[parent[0].id] = parent[0];
+			}
+		}
+		// This is unfortunately 90x (!) faster than any of ES6 syntactic sugar
+		let result = [];
+		for (let id in itemMap) {
+			result.push(itemMap[id])
+		}
+			
+		return result;	
+	}
+	
+	
```

Comment:
I don't think this the behavior we want. Dragging a child item shouldn't work, and I don't have a problem with just disallowing the drag in those cases. What #36 specifies is allowing the drag if both the parent item and all child items are selected (as you would get if you selected an range with a parent expanded). To account for the case of a search, where some child items were in gray, we should probably allow for any number of child items of selected parents. But I don't think we don't want to allow dragging of only child items, because that's confusing.